libsolidity/formal/PredicateInstance.cpp

```diff
@@ -128,9 +164,29 @@ vector<smtutil::Expression> currentFunctionVariables(
 	return exprs;
 }
 
+vector<smtutil::Expression> currentFunctionVariablesForCall(
+	FunctionDefinition const& _function,
+	ContractDefinition const* _contract,
+	EncodingContext& _context
+)
+{
+	auto& state = _context.state();
+	vector<smtutil::Expression> exprs{_context.state().errorFlag().currentValue(), state.thisAddress(0), state.crypto(0), state.tx(0), state.state()};
```

Comment:
Maybe split into more lines, plus use state in first argument as well.
```suggestion
	vector<smtutil::Expression> exprs{
		state.errorFlag().currentValue(), 
		state.thisAddress(0), 
		state.crypto(0), 
		state.tx(0), 
		state.state()
	};
```

Maybe one more idea is to create a method in state so that it returns list of expressions for this case.

Reply:
The annoying part is it's not always the same subset that we're interested in. For example, here for the last one we want `state.state()`, but in the next function we want `state.state(0)`

Reply:
I see :)